lib/practice_exercises.rb

```diff
@@ -1,13 +1,42 @@
+# Time Complexity: linear
+# Space Complexity: linear
 
-# Time Complexity: ?
-# Space Complexity: ?
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  # remove white spaces
+  string = string.gsub(/\s+/, "").downcase
```

Comment:
You could do something similar here to remove punctuation as well.

```suggestion
  string = string.gsub(/[^a-z^A-Z]/, "").downcase
```